File: work/pmod_swt/pmod_swt.py
from gpio import *   # Importe tous les éléments de la bibliothèque gpio
import sys          # Importe le module sys pour gérer les arguments en ligne de commande

NB_PINS = 8         # Définit le nombre de broches à lire

if len(sys.argv) < 2:   # Si aucun argument en ligne de commande n'a été fourni
    print(f"Usage: {sys.argv[0]} <port>")   # Affiche un message d'erreur avec le nom du script
    sys.exit(1)         # Arrête le script avec un code d'erreur

port = sys.argv[1]  # Récupère le premier argument en ligne de commande
if port == 'a':     # Si le port spécifié est 'a'
    pin_list = JA   # Utilise la liste de broches PA
elif port == 'b':   # Sinon, si le port spécifié est 'b'
    pin_list = JB   # Utilise la liste de broches PB
elif port == 'c':   # Sinon, si le port spécifié est 'c'
    pin_list = JC   # Utilise la liste de broches PC
else:               # Sinon, si le port spécifié n'est ni 'a', ni 'b', ni 'c'
    print(f"Port inconnu : {port}")    # Affiche un message d'erreur
    sys.exit(1)         # Arrête le script avec un code d'erreur

val = 0           # Initialise la variable qui stockera la valeur binaire lue
bit_count = 0       # Initialise le compteur de bits lus

for i in range(NB_PINS):    # Boucle sur toutes les broches à lire
    gpio_set_direction(pin_list[i], 0)   # Configure la broche en entrée
    bit = gpio_get_value(pin_list[i])    # Lit la valeur de la broche
    val += bit * (1 << bit_count)      # Convertit la valeur lue en nombre binaire
    print(f"{pin_list[i]} : {bit}")     # Affiche la valeur lue pour la broche
    bit_count += 1      # Incrémente le compteur de bits lus

print(f"{val:x}") 

File: work/pmod_swt/pmod_swt.c
#include "/home/mi005558/pi/work/common/common.h"
#include "/home/mi005558/pi/work/gpio/gpio.h"
#include <stdio.h>
#include <stdlib.h>

#define count 8

int main(int argc, char *argv[]) {
    // Vérifie que le port GPIO est spécifié en argument
    if (argc < 2) {
        printf("Usage: %s <port>\n", argv[0]);
        return 1;
    }

    // Définit une variable pour le port GPIO spécifié
    char *Port = argv[1];
    char* J;
	
	// Assigne le tableau de broches correspondant au port GPIO spécifié
    if(Port[0] == 'a'){
        J = JA;
    }else if(Port[0] == 'b'){
        J = JB;
    }else if(Port[0] == 'c'){
        J = JC;
    }else{
        // Affiche une erreur si le port n'est pas reconnu
        fprintf(stderr, "le port est introuvable : %c\n", Port[0]);
        return 1;
    }

    int val = 0;
    int bcount = 0;

    // Parcourt les broches du port GPIO et récupère leur état
    for (int i = 0; i < count; i++) {
        gpio_set_direction(J[i], 0);
        int Bit = gpio_get_value(J[i]);

        // Ajoute le bit récupéré à la valeur totale
        val += Bit * (1 << bcount);
        printf("%d : %d\n", J[i], Bit);
        bcount++;
    }

    // Affiche la valeur récupérée en format hexadécimal
    printf("%x\n", val);

    return 0;
}
